### PR TITLE
Added the possibility to create self signed certificates for applicat…

### PR DESCRIPTION
**Pull request by apichick**
Added the possibility to create self signed certificates for applicat…

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/net-lb-app-ext/self-signed-certs.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "tls_private_key" "private_keys" {
+  for_each  = toset(var.ssl_certificates.self_signed_configs)
+  algorithm = "RSA"
+  rsa_bits  = 4096
```
Comment (wiktorn) [inline modules/net-lb-app-ext/self-signed-certs.tf:20]: We only support [RSA-2048 or ECDSA P-256](https://cloud.google.com/load-balancing/docs/ssl-certificates/self-managed-certs#create-key-and-cert) encryption algorithms.

**Review by juliocc — COMMENTED** [review]
I'm on the fence with this PR. To me, creating a certificate feels more like an example than something to be handled by the LB modules.

File: modules/net-lb-app-ext/self-signed-certs.tf (hunk 1 above)
Comment (apichick) [reply modules/net-lb-app-ext/self-signed-certs.tf:20]: ah true, let me change that

Comment (apichick) [reply modules/net-lb-app-ext/self-signed-certs.tf]: done

**Comment by ludoo** [comment]
> I'm on the fence with this PR. To me, creating a certificate feels more like an example than something to be handled by the LB modules.

I also don't like having this in the LB modules. But +1 on showing how to do it in one example.

**Comment by ludoo** [comment]
Miren, I'm closing this let's come up with an example instead. :)

**Pull request closed without merging**